File: FruitWholesale.Client/src/app/features/employee-work-log/employee-work-log.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { PaginatedList, PaginationRequest } from '../../core/models/common.model';
import { EmployeeWorkLog, SaveEmployeeWorkLog } from '../../core/models/master-data.model';
import { toHttpParams } from '../../core/utils/http-params.util';

@Injectable({ providedIn: 'root' })
export class EmployeeWorkLogService {
  private readonly baseUrl = `${environment.apiUrl}/employeeworklog`;

  constructor(private readonly http: HttpClient) {}

  getPaged(request: PaginationRequest, employeeId?: number | null, fromDate?: string | null, toDate?: string | null) {
    return this.http.get<PaginatedList<EmployeeWorkLog>>(this.baseUrl, {
      params: toHttpParams({ ...request, employeeId, fromDate, toDate })
    });
  }

  create(dto: SaveEmployeeWorkLog): Observable<EmployeeWorkLog> {
    return this.http.post<EmployeeWorkLog>(this.baseUrl, dto);
  }

  update(id: number, dto: SaveEmployeeWorkLog): Observable<EmployeeWorkLog> {
    return this.http.put<EmployeeWorkLog>(`${this.baseUrl}/${id}`, dto);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}

File: FruitWholesale.Mobile/lib/features/employee_work_log/employee_work_log_service.dart
import '../../core/api/api_client.dart';
import '../../core/models/paginated_list.dart';
import 'employee_work_log_models.dart';

class EmployeeWorkLogService {
  final ApiClient _api;

  EmployeeWorkLogService(this._api);

  Future<PaginatedList<EmployeeWorkLog>> getPaged({
    int pageNumber = 1,
    int pageSize = 20,
    int? employeeId,
    String? fromDate,
    String? toDate,
  }) async {
    final json = await _api.get('/employeeworklog', query: {
      'pageNumber': pageNumber,
      'pageSize': pageSize,
      if (employeeId != null) 'employeeId': employeeId,
      if (fromDate != null) 'fromDate': fromDate,
      if (toDate != null) 'toDate': toDate,
    });
    return PaginatedList.fromJson(json as Map<String, dynamic>, EmployeeWorkLog.fromJson);
  }

  Future<EmployeeWorkLog> getById(int id) async {
    final json = await _api.get('/employeeworklog/$id');
    return EmployeeWorkLog.fromJson(json as Map<String, dynamic>);
  }

  Future<EmployeeWorkLog> create(EmployeeWorkLog log) async {
    final json = await _api.post('/employeeworklog', body: log.toSaveJson());
    return EmployeeWorkLog.fromJson(json as Map<String, dynamic>);
  }

  Future<EmployeeWorkLog> update(int id, EmployeeWorkLog log) async {
    final json = await _api.put('/employeeworklog/$id', body: log.toSaveJson());
    return EmployeeWorkLog.fromJson(json as Map<String, dynamic>);
  }
}
